src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -236,82 +252,101 @@ private void filmSuchen1(String strUrlFeed, String thema, boolean weiter) {
+                    zeit = zeit + ":00";
+                    datum = datum.substring(0, datum.indexOf("|")).trim();
+                }
+                titel = seite1.extract("<h4 class=\"headline\">", "<", pos);
+                dauer = seite1.extract("<p class=\"subtitle\">", "<", pos);
+                try
+                {
+                    dauer = dauer.replace("Min.", "").trim();
+                    dauer = dauer.replace("| UT", "").trim();
+                    d = Long.parseLong(dauer) * 60;
+                } catch (Exception ex)
+                {
+                }
+                if (d == 0)
+                {
+                    Log.errorLog(915263621, "Dauer==0: " + strUrlFeed);
+                }
+                urlSendung = seite1.extract("<a href=\"/tv/", "\"", pos);
+                if (!urlSendung.isEmpty())
+                {
+                    urlSendung = "http://www.ardmediathek.de/tv/" + urlSendung;
+                    urlSendung = urlSendung.replace("&amp;", "&");
                 }
+
+                filmSuchen2(url, thema, titel, d, datum, zeit, urlSendung);
             }
             if (!Config.getStop() && weiter
-                    && (CrawlerTool.loadLongMax() || thema.equalsIgnoreCase("alpha-Centauri"))) {
+                    && (CrawlerTool.loadLongMax() || thema.equalsIgnoreCase("alpha-Centauri")))
+            {
                 // dann gehts weiter
                 int maxWeiter = 0;
                 int maxTh = 10;
                 String urlWeiter = strUrlFeed + "&mcontents=page.";
-                for (int i = 2; i < maxTh; ++i) {
+                for (int i = 2; i < maxTh; ++i)
+                {
                     ///tv/Abendschau/Sendung?documentId=14913430&amp;bcastId=14913430&amp;mcontents=page.2"
-                    if (seite1.indexOf("&amp;mcontents=page." + i) != -1) {
+                    if (seite1.indexOf("&amp;mcontents=page." + i) != -1)
+                    {
                         maxWeiter = i;
-                    } else {
+                    } else
+                    {
                         break;
                     }
                 }
-                for (int i = 2; i < maxTh; ++i) {
-                    if (Config.getStop()) {
+                for (int i = 2; i < maxTh; ++i)
+                {
+                    if (Config.getStop())
+                    {
                         break;
                     }
-                    if (i <= maxWeiter) {
+                    if (i <= maxWeiter)
+                    {
                         filmSuchen1(urlWeiter + i, thema, false);
-                    } else {
+                    } else
+                    {
                         break;
                     }
 
                 }
             }
-        }
+}
 
         private void filmSuchen2(String urlFilm_, String thema, String titel, long dauer, String datum, String zeit, String urlSendung) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 51 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00117)